### PR TITLE
Mc dl genericiserer 4368

### PR DESCRIPTION
**Pull request by kingmushroom**
making generic. Tested on aws(happy to show) , with both sandy and blocks lookups n data.

### CODE REVIEW

File: tests/test_enrichment.py [hunk 1]
```diff
@@ -342,61 +324,65 @@ def test_marine_mismatch_detector(self):
...
+                 "0": {"filename": "responderlookup",
+                       "columnstokeep": ["responder_id", "county"],
+                       "joincolumn": "responder_id",
+                       "required": ["county"]},
+                 "1": {"filename": "countylookup",
+                       "columnstokeep": ["county_name",
+                                         "region", "county",
+                                         "marine"],
+                       "joincolumn": "county",
+                       "required": ["region", "marine"]}},
```
Comment (KrisRogos) [inline tests/test_enrichment.py]: Column names are not separated with `_`, `-` or differentCase

File: tests/test_enrichment.py [hunk 2]
```diff
@@ -444,8 +418,23 @@ def test_lambder_handler(self):
...
+            input = {"data": testdata, "lookups": {
+                "0": {"filename": "responderlookup",
+                      "columnstokeep": ["responder_id", "county"],
+                      "joincolumn": "responder_id",
+                      "required": ["county"]},
+                "1": {"filename": "countylookup",
+                      "columnstokeep": ["county_name",
+                                        "region", "county",
+                                        "marine"],
+                      "joincolumn": "county",
```
Comment (KrisRogos) [inline tests/test_enrichment.py]: Same as above

File: tests/test_enrichment.py [hunk 3]
```diff
@@ -498,23 +487,35 @@ def test_method_client_error(self):
...
         ):
+            with open("tests/fixtures/test_data.json", "r") as file:
+                testdata = file.read()
+            parameters = {"marine_mismatch_check": "true", "period_column": "period",
+                          "identifier_column": "responder_id"}
+            input = {"data": testdata, "lookups":
+                     {"0": {"required": "yup",
+                      "filename": "mike",
+                            "columnstokeep": "moo",
+                            "joincolumn": "fred"}}, "parameters": parameters}
```
Comment (KrisRogos) [inline tests/test_enrichment.py]: And again

**Review by KrisRogos — CHANGES_REQUESTED** [review]
The following sections don't match the current naming convention. Everything else is fine.

**Review by KrisRogos — APPROVED** [review]

**Review by glanvl — APPROVED** [review]
Looks all fine to me. Tests ran fine locally and also tested on AWS with triggering the wrangler. All worked fine and produced the output with the required columns. The only possible change which I would suggest (could be left as tech debt) would be to have the JSON lookup "Lookup_info" environment variable into a file as a separate config.

**Pull request merged**